lib/ProductOpener/APITest.pm

```diff
@@ -124,4 +124,31 @@ sub create_user ($ua, $args_ref) {
 	return;
 }
 
+sub create_product ($ua, $args_ref) {
+	
+	my %fields = (
```

Comment:
I would remove all the product specific data (from "code" to "packaging_text"), this data should be provided by callers of the function.

e.g. when in the test file you do:

create_product($ua, {});

# edit preference accessible
my $response = $ua->get("http://frontend/cgi/product.pl?type=edit&code=2000000000099");

Anything you test for should be defined in the arguments of the create_product() call, so that's it's possible to understand the test file. Here for instance, it's not obvious that the code 2000000000099 is set by default by create_product()

Reply:
@stephanegigandet thank you! i committed the suggestions